include/llbuild/Core/BuildEngine.h

```diff
@@ -66,6 +66,9 @@ struct Result {
   // FIXME: At some point, figure out the optimal representation for this field,
   // which is likely to be a lot of the resident memory size.
   std::vector<KeyID> dependencies;
+
+  /// Is this result transient.
+  bool transient = false;
```

Comment:
This makes me nervous. For one thing, it feels like it exposes a potential behavioral difference when running with a cached vs uncached engine. More generally, I am nervous about introducing more semantic options in the core, without a clear understanding of their meaning. It *feels* to me like we should be able to solve this problem without this, which makes me thing this is introducing a semantic behavior that isn't actually necessary (and may be hard to prove correct).

Reply:
Moving to a separate PR.